packages/js/src/httpClient.ts

```diff
@@ -44,17 +44,27 @@ export interface ClientOptions {
   orgId?: string;
 }
 
-// The browsers don't have process.env, fake it
-const process: { env: Record<string, string | undefined> } =
-  typeof window === 'undefined' ? global.process : { env: {} };
-
 export default abstract class HTTPClient {
   protected readonly client: FetchClient;
 
   constructor(options: ClientOptions = {}) {
-    const token = options.token || process.env.AXIOM_TOKEN || '';
-    const url = options.url || process.env.AXIOM_URL || AxiomURL;
-    const orgId = options.orgId || process.env.AXIOM_ORG_ID || '';
+    let token = options.token;
+    if (!token && !isBrowser) {
+      token = process.env.AXIOM_TOKEN || '';
+    } else if (!token && isBrowser) {
+      console.warn('must provide Axiom token')
+    }
+    let url = options.url;
+    if (!url && !isBrowser) {
+      url = process.env.AXIOM_URL || AxiomURL;
+    } else if (!url && isBrowser) {
+      url = AxiomURL;
+    }
+
+    let orgId = options.orgId;
+    if (!orgId && !isBrowser) {
+      orgId = process.env.AXIOM_ORG_ID || ''
+    }
```

Comment:
That means that if there's a pkg which exports environment variables (like Next.js does with NEXT_PUBLIC) it won't pick it up, right?

Reply:
hmm that's right, I am leaning towards customers providing the credentials instead of automatic pickup, but anyway in case of NEXT_PUBLIC_* user would have to provide those because we don't check for NEXT_PUBLIC since its a generic lib.